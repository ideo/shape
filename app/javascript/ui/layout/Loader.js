import PropTypes from 'prop-types'
import { CubeGrid } from 'styled-loaders-react'
import { Flex, Box } from 'reflexbox'

import v from '~/utils/variables'

const CircularLoader = ({ color }) => (
  <svg
    version="1.0"
    width="64px"
    height="64px"
    viewBox="0 0 128 128"
    xmlSpace="preserve"
  >
    <path
      fill={color}
      fillOpacity="1"
      d="M64.4 16a49 49 0 0 0-50 48 51 51 0 0 0 50 52.2 53 53 0 0 0 54-52c-.7-48-45-55.7-45-55.7s45.3 3.8 49 55.6c.8 32-24.8 59.5-58 60.2-33 .8-61.4-25.7-62-60C1.3 29.8 28.8.6 64.3 0c0 0 8.5 0 8.7 8.4 0 8-8.6 7.6-8.6 7.6z"
    >
      <animateTransform
        attributeName="transform"
        type="rotate"
        from="0 64 64"
        to="360 64 64"
        dur="1600ms"
        repeatCount="indefinite"
      ></animateTransform>
    </path>
  </svg>
)

CircularLoader.propTypes = {
  color: PropTypes.string.isRequired,
}

class Loader extends React.PureComponent {
  render() {
    const { containerHeight, animation, size } = this.props
    // note "circular" SVG does not respect "size" prop
    const LoaderAnimation = animation === 'circular' ? CircularLoader : CubeGrid
    return (
      <Flex style={{ height: containerHeight }} align="center" justify="center">
        <Box>
          <LoaderAnimation color={v.colors.commonDark} size={`${size}px`} />
        </Box>
      </Flex>
    )
  }
}

Loader.propTypes = {
  containerHeight: PropTypes.string,
  size: PropTypes.number,
  animation: PropTypes.oneOf(['cube', 'circular']),
}
Loader.defaultProps = {
  containerHeight: '50vh',
  size: 100,
  animation: 'cube',
}

export default Loader
