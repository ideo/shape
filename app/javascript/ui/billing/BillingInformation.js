import { Fragment } from 'react'
import { observable, runInAction } from 'mobx'
import Grid from '@material-ui/core/Grid'
import Hidden from '@material-ui/core/Hidden'
import PropTypes from 'prop-types'
import { inject, observer, PropTypes as MobxPropTypes } from 'mobx-react'
import styled from 'styled-components'
import Section from '~shared/components/molecules/Section'
import Typography from '~shared/components/atoms/Typography'
import FancyDollarAmount from '~shared/components/atoms/FancyDollarAmount'
import Box from '~shared/components/atoms/Box'
import HorizontalDivider from '~shared/components/atoms/HorizontalDivider'
import Loader from '~/ui/layout/Loader'
import trackError from '~/utils/trackError'
import {
  formatAsDollarAmount,
  dateParseISO,
  formatDate,
} from '~shared/utils/formatters'
import v from '~/utils/variables'

const Wrapper = styled.div`
  font-family: 'Gotham';
`

const Instructional = styled.div`
  font-size: 0.75rem;
  line-height: 16px;
  color: ${v.colors.commonMedium};
  letter-spacing: 0;
`

const BillingNoticeWrapper = styled.div`
  text-align: center;
`

const BillingNotice = styled.div`
  font-size: 1rem;
  line-height: 22px;
  color: #a89f9b;
  max-width: 500px;
  margin: 0 auto 20px;
`

const BlockHeader = styled.h3`
  font-size: 1rem;
  letter-spacing: 0.5px;
  font-weight: 500;

  @media only screen and (max-width: ${v.responsive.muiSmBreakpoint}px) {
    margin-bottom: 0.25rem;
  }
`

const Block = ({ title, children }) => (
  <Grid item sm={12} md={4} lg={3}>
    <BlockHeader>{title}</BlockHeader>
    <Box mt={1}>{children}</Box>
    <Hidden mdUp>
      {/* add spacing once the blocks stack at small sizes */}
      <Box mb={24} />
    </Hidden>
  </Grid>
)
Block.propTypes = {
  title: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
}

const TrialHighlight = styled.div`
  background-color: #c0dbde;
`

const FreeTrial = styled(TrialHighlight)`
  font-size: 1rem;
  line-height: 11px;
  letter-spacing: 0.5px;
  padding: 8px;
  margin: 15px 0;
  font-weight: 500;
  display: inline-block;
`

const Label = styled.span`
  font-size: 1rem;
  color: ${v.colors.commonDark};
`

@inject('apiStore', 'networkStore')
@observer
class BillingInformation extends React.Component {
  @observable
  loaded = false

  componentWillMount() {
    this.load()
  }

  async load() {
    const {
      apiStore: { currentUserOrganizationId: organizationId },
      networkStore,
    } = this.props
    try {
      await networkStore.loadOrganization(organizationId)
      await networkStore.loadPlans(organizationId)
      runInAction(() => (this.loaded = true))
    } catch (e) {
      trackError(e)
    }
  }

  render() {
    if (!this.loaded) {
      return <Loader />
    }

    const {
      currentUserOrganization: {
        in_app_billing,
        active_users_count,
        current_billing_period_end,
        current_billing_period_start,
        is_within_trial_period,
        name,
        price_per_user,
        trial_ends_at,
        trial_users_count,
        deactivated,
      },
    } = this.props.apiStore

    if (deactivated) {
      return null
    }

    const trialsUsedCount =
      active_users_count > trial_users_count
        ? trial_users_count
        : active_users_count

    const billableUserCount = is_within_trial_period
      ? active_users_count - trial_users_count
      : active_users_count

    const currentMonthlyRate =
      billableUserCount > 0 ? billableUserCount * price_per_user : 0

    const formatISODate = d => formatDate(dateParseISO(d), 'MM/dd/yyyy')

    return (
      <Wrapper>
        <Section title="Billing Information">
          <Grid container justify="space-between">
            <Block title="Current Active People">
              <Grid container justify="space-between" alignItems="flex-end">
                <Grid item>
                  <Typography variant="emphasis">
                    {active_users_count}
                  </Typography>
                </Grid>
                <Grid item>{is_within_trial_period && 'Total People'}</Grid>
              </Grid>
              {is_within_trial_period && (
                <div>
                  <TrialHighlight>
                    <Box my={20} p={10}>
                      <Grid container justify="space-between">
                        <Grid item>{trialsUsedCount}</Grid>
                        <Grid item>People on free trial</Grid>
                      </Grid>
                    </Box>
                  </TrialHighlight>
                  <HorizontalDivider />
                  <Box p={10}>
                    <Grid container justify="space-between">
                      <Grid item>
                        {billableUserCount < 1 ? 0 : billableUserCount}
                      </Grid>
                      <Grid item>Billable people</Grid>
                    </Grid>
                  </Box>
                </div>
              )}
            </Block>
            {in_app_billing ? (
              <Fragment>
                <Block title="Current Monthly Rate">
                  <FancyDollarAmount>{currentMonthlyRate}</FancyDollarAmount>
                  <Box my={25}>
                    {formatAsDollarAmount(price_per_user)}
                    /person per month
                  </Box>
                  <Instructional>
                    Total monthly rate does not include people covered by free
                    trial conditions.
                  </Instructional>
                </Block>
                <Block title="Current Billing Period">
                  <Grid container justify="space-between" spacing={16}>
                    <Grid item>
                      <Label>Start date:</Label>
                    </Grid>
                    <Grid item>
                      {formatISODate(current_billing_period_start)}
                    </Grid>
                  </Grid>
                  <Grid container justify="space-between" spacing={16}>
                    <Grid item>
                      <Label>End date:</Label>
                    </Grid>
                    <Grid item>
                      {formatISODate(current_billing_period_end)}
                    </Grid>
                  </Grid>
                  {is_within_trial_period && (
                    <div>
                      <FreeTrial>FREE TRIAL</FreeTrial>
                      <Grid container justify="space-between" spacing={16}>
                        <Grid item>
                          <Label>Free trials used:</Label>
                        </Grid>
                        <Grid item>
                          {trialsUsedCount} of {trial_users_count}
                        </Grid>
                      </Grid>
                      <Grid container justify="space-between" spacing={16}>
                        <Grid item>
                          <Label>Trial expires:</Label>
                        </Grid>
                        <Grid item>{formatISODate(trial_ends_at)}</Grid>
                      </Grid>
                    </div>
                  )}
                </Block>
              </Fragment>
            ) : (
              <Grid item xs={9}>
                <BillingNoticeWrapper>
                  <Box my={20}>
                    <h3>Current Rate & Billing Period</h3>
                  </Box>
                  <BillingNotice>
                    Billing for {name} instance is not handled by this
                    application. Email{' '}
                    <a href="mailto:hello@shape.space">hello@shape.space</a> for
                    more information.
                  </BillingNotice>
                </BillingNoticeWrapper>
              </Grid>
            )}
          </Grid>
        </Section>
      </Wrapper>
    )
  }
}

BillingInformation.wrappedComponent.propTypes = {
  apiStore: MobxPropTypes.objectOrObservableObject.isRequired,
  networkStore: MobxPropTypes.objectOrObservableObject.isRequired,
}

export default BillingInformation
