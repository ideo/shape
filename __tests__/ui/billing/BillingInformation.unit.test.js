import BillingInformation from '~/ui/billing/BillingInformation'
import { shallow } from 'enzyme'
import Loader from '~/ui/layout/Loader'
import expectTreeToMatchSnapshot from '#/helpers/expectTreeToMatchSnapshot'

const render = async (apiStore, networkStore) => {
  const wrapper = shallow(
    <BillingInformation.wrappedComponent
      apiStore={apiStore}
      networkStore={networkStore}
    />
  )

  expect(wrapper.find(Loader).exists()).toBe(true)

  await Promise.all([networkStore.loadOrganization, networkStore.loadPlans])
  wrapper.update()
  expect(wrapper.find(Loader).exists()).toBe(false)
  expect(networkStore.loadOrganization).toHaveBeenCalledWith(
    apiStore.currentUserOrganizationId
  )
  expect(networkStore.loadPlans).toHaveBeenCalledWith(
    apiStore.currentUserOrganizationId
  )
  expectTreeToMatchSnapshot(wrapper)
}

describe('BillingInformation', () => {
  let apiStore, networkStore

  beforeEach(() => {
    apiStore = {
      currentUserOrganizationId: 123,
      currentUserOrganization: {
        active_users_count: 0,
        trial_users_count: 0,
        trial_ends_at: '2022-01-01',
        is_within_trial_period: true,
        price_per_user: 1.23,
        current_billing_period_start: '2015-01-01',
        current_billing_period_end: '2015-01-31',
        deactivated: false,
      },
    }
    networkStore = {
      loadOrganization: jest.fn(() => Promise.resolve()),
      loadPlans: jest.fn(() => Promise.resolve()),
    }
  })

  describe('organization is deactivated', () => {
    beforeEach(() => {
      apiStore.currentUserOrganization.deactivated = true
    })

    it('renders nothing', async () => {
      await render(apiStore, networkStore)
    })
  })

  describe('in app billing enabled', () => {
    beforeEach(() => {
      apiStore.currentUserOrganization.in_app_billing = true
    })
    describe('still within trial', () => {
      beforeEach(() => {
        apiStore.currentUserOrganization.is_within_trial_period = true
        apiStore.currentUserOrganization.trial_users_count = 10
        apiStore.currentUserOrganization.active_users_count = 5
      })
      it('renders trial information', async () => {
        await render(apiStore, networkStore)
      })
    })

    describe('trial expired', () => {
      beforeEach(() => {
        apiStore.currentUserOrganization.is_within_trial_period = false
        apiStore.currentUserOrganization.active_users_count = 20
      })

      it('renders without trial information', async () => {
        await render(apiStore, networkStore)
      })
    })

    describe('within trial, exceeded trial users', () => {
      beforeEach(() => {
        apiStore.currentUserOrganization.is_within_trial_period = true
        apiStore.currentUserOrganization.trial_users_count = 5
        apiStore.currentUserOrganization.active_users_count = 100
      })

      it('renders with trial and billing information', async () => {
        await render(apiStore, networkStore)
      })
    })
  })

  describe('in app billing disabled', () => {
    beforeEach(() => {
      apiStore.currentUserOrganization.in_app_billing = true
    })

    it('renders with the in app billing notice', async () => {
      await render(apiStore, networkStore)
    })
  })
})
