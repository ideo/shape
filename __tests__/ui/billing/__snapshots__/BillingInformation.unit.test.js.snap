// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BillingInformation in app billing disabled does not render the trial information 1`] = `
<styled.div>
  <Section
    title="Billing Information"
  >
    <WithStyles(Grid)
      container={true}
      justify="space-between"
    >
      <Block
        title="Current Active People"
      >
        <WithStyles(Grid)
          alignItems="flex-end"
          container={true}
          justify="space-between"
        >
          <WithStyles(Grid)
            item={true}
          >
            <Typography
              variant="emphasis"
            >
              0
            </Typography>
          </WithStyles(Grid)>
          <WithStyles(Grid)
            item={true}
          >
            Total People
          </WithStyles(Grid)>
        </WithStyles(Grid)>
        <div>
          <styled.div>
            <Box
              my={20}
              p={10}
            >
              <WithStyles(Grid)
                container={true}
                justify="space-between"
              >
                <WithStyles(Grid)
                  item={true}
                >
                  0
                </WithStyles(Grid)>
                <WithStyles(Grid)
                  item={true}
                >
                  People on free trial
                </WithStyles(Grid)>
              </WithStyles(Grid)>
            </Box>
          </styled.div>
          <HorizontalDivider />
          <Box
            p={10}
          >
            <WithStyles(Grid)
              container={true}
              justify="space-between"
            >
              <WithStyles(Grid)
                data-jest="billableUserCount"
                item={true}
              >
                0
              </WithStyles(Grid)>
              <WithStyles(Grid)
                item={true}
              >
                Billable people
              </WithStyles(Grid)>
            </WithStyles(Grid)>
          </Box>
        </div>
      </Block>
      <WithStyles(Grid)
        item={true}
        xs={9}
      >
        <styled.div>
          <Box
            my={20}
          >
            <h3>
              Current Rate & Billing Period
            </h3>
          </Box>
          <styled.div>
            Billing for 
             instance is not handled by this application. Email
             
            <a
              href="mailto:hello@shape.space"
            >
              hello@shape.space
            </a>
             for more information.
          </styled.div>
        </styled.div>
      </WithStyles(Grid)>
    </WithStyles(Grid)>
  </Section>
</styled.div>
`;

exports[`BillingInformation in app billing disabled renders with the in app billing notice 1`] = `
<styled.div>
  <Section
    title="Billing Information"
  >
    <WithStyles(Grid)
      container={true}
      justify="space-between"
    >
      <Block
        title="Current Active People"
      >
        <WithStyles(Grid)
          alignItems="flex-end"
          container={true}
          justify="space-between"
        >
          <WithStyles(Grid)
            item={true}
          >
            <Typography
              variant="emphasis"
            >
              0
            </Typography>
          </WithStyles(Grid)>
          <WithStyles(Grid)
            item={true}
          >
            Total People
          </WithStyles(Grid)>
        </WithStyles(Grid)>
        <div>
          <styled.div>
            <Box
              my={20}
              p={10}
            >
              <WithStyles(Grid)
                container={true}
                justify="space-between"
              >
                <WithStyles(Grid)
                  item={true}
                >
                  0
                </WithStyles(Grid)>
                <WithStyles(Grid)
                  item={true}
                >
                  People on free trial
                </WithStyles(Grid)>
              </WithStyles(Grid)>
            </Box>
          </styled.div>
          <HorizontalDivider />
          <Box
            p={10}
          >
            <WithStyles(Grid)
              container={true}
              justify="space-between"
            >
              <WithStyles(Grid)
                data-jest="billableUserCount"
                item={true}
              >
                0
              </WithStyles(Grid)>
              <WithStyles(Grid)
                item={true}
              >
                Billable people
              </WithStyles(Grid)>
            </WithStyles(Grid)>
          </Box>
        </div>
      </Block>
      <WithStyles(Grid)
        item={true}
        xs={9}
      >
        <styled.div>
          <Box
            my={20}
          >
            <h3>
              Current Rate & Billing Period
            </h3>
          </Box>
          <styled.div>
            Billing for 
             instance is not handled by this application. Email
             
            <a
              href="mailto:hello@shape.space"
            >
              hello@shape.space
            </a>
             for more information.
          </styled.div>
        </styled.div>
      </WithStyles(Grid)>
    </WithStyles(Grid)>
  </Section>
</styled.div>
`;

exports[`BillingInformation in app billing enabled still within trial renders trial information 1`] = `
<styled.div>
  <Section
    title="Billing Information"
  >
    <WithStyles(Grid)
      container={true}
      justify="space-between"
    >
      <Block
        title="Current Active People"
      >
        <WithStyles(Grid)
          alignItems="flex-end"
          container={true}
          justify="space-between"
        >
          <WithStyles(Grid)
            item={true}
          >
            <Typography
              variant="emphasis"
            >
              5
            </Typography>
          </WithStyles(Grid)>
          <WithStyles(Grid)
            item={true}
          >
            Total People
          </WithStyles(Grid)>
        </WithStyles(Grid)>
        <div>
          <styled.div>
            <Box
              my={20}
              p={10}
            >
              <WithStyles(Grid)
                container={true}
                justify="space-between"
              >
                <WithStyles(Grid)
                  item={true}
                >
                  5
                </WithStyles(Grid)>
                <WithStyles(Grid)
                  item={true}
                >
                  People on free trial
                </WithStyles(Grid)>
              </WithStyles(Grid)>
            </Box>
          </styled.div>
          <HorizontalDivider />
          <Box
            p={10}
          >
            <WithStyles(Grid)
              container={true}
              justify="space-between"
            >
              <WithStyles(Grid)
                data-jest="billableUserCount"
                item={true}
              >
                0
              </WithStyles(Grid)>
              <WithStyles(Grid)
                item={true}
              >
                Billable people
              </WithStyles(Grid)>
            </WithStyles(Grid)>
          </Box>
        </div>
      </Block>
      <Block
        title="Current Monthly Rate"
      >
        <FancyDollarAmount>
          0
        </FancyDollarAmount>
        <Box
          my={25}
        >
          $5.00
          /person per month
        </Box>
        <styled.div>
          Total monthly rate does not include people covered by free trial conditions.
        </styled.div>
      </Block>
      <Block
        title="Current Billing Period"
      >
        <WithStyles(Grid)
          container={true}
          justify="space-between"
          spacing={16}
        >
          <WithStyles(Grid)
            item={true}
          >
            <styled.span>
              Start date:
            </styled.span>
          </WithStyles(Grid)>
          <WithStyles(Grid)
            item={true}
          >
            01/01/2015
          </WithStyles(Grid)>
        </WithStyles(Grid)>
        <WithStyles(Grid)
          container={true}
          justify="space-between"
          spacing={16}
        >
          <WithStyles(Grid)
            item={true}
          >
            <styled.span>
              End date:
            </styled.span>
          </WithStyles(Grid)>
          <WithStyles(Grid)
            item={true}
          >
            01/31/2015
          </WithStyles(Grid)>
        </WithStyles(Grid)>
        <FreeTrial
          trialEndsAt="01/01/2022"
          trialLabel="Free trials used"
          trialTitle="FREE TRIAL"
          trialUsersCount={10}
          trialsUsedCount={5}
        />
      </Block>
    </WithStyles(Grid)>
  </Section>
</styled.div>
`;

exports[`BillingInformation in app billing enabled trial expired renders without trial information 1`] = `
<styled.div>
  <Section
    title="Billing Information"
  >
    <WithStyles(Grid)
      container={true}
      justify="space-between"
    >
      <Block
        title="Current Active People"
      >
        <WithStyles(Grid)
          alignItems="flex-end"
          container={true}
          justify="space-between"
        >
          <WithStyles(Grid)
            item={true}
          >
            <Typography
              variant="emphasis"
            >
              20
            </Typography>
          </WithStyles(Grid)>
          <WithStyles(Grid)
            item={true}
          />
        </WithStyles(Grid)>
      </Block>
      <Block
        title="Current Monthly Rate"
      >
        <FancyDollarAmount>
          100
        </FancyDollarAmount>
        <Box
          my={25}
        >
          $5.00
          /person per month
        </Box>
        <styled.div>
          Total monthly rate does not include people covered by free trial conditions.
        </styled.div>
      </Block>
      <Block
        title="Current Billing Period"
      >
        <WithStyles(Grid)
          container={true}
          justify="space-between"
          spacing={16}
        >
          <WithStyles(Grid)
            item={true}
          >
            <styled.span>
              Start date:
            </styled.span>
          </WithStyles(Grid)>
          <WithStyles(Grid)
            item={true}
          >
            01/01/2015
          </WithStyles(Grid)>
        </WithStyles(Grid)>
        <WithStyles(Grid)
          container={true}
          justify="space-between"
          spacing={16}
        >
          <WithStyles(Grid)
            item={true}
          >
            <styled.span>
              End date:
            </styled.span>
          </WithStyles(Grid)>
          <WithStyles(Grid)
            item={true}
          >
            01/31/2015
          </WithStyles(Grid)>
        </WithStyles(Grid)>
      </Block>
    </WithStyles(Grid)>
  </Section>
</styled.div>
`;

exports[`BillingInformation in app billing enabled with freemium users are over the limit does not render the trial information 1`] = `
<styled.div>
  <Section
    title="Billing Information"
  >
    <WithStyles(Grid)
      container={true}
      justify="space-between"
    >
      <Block
        title="Current Active People"
      >
        <WithStyles(Grid)
          alignItems="flex-end"
          container={true}
          justify="space-between"
        >
          <WithStyles(Grid)
            item={true}
          >
            <Typography
              variant="emphasis"
            >
              6
            </Typography>
          </WithStyles(Grid)>
          <WithStyles(Grid)
            item={true}
          />
        </WithStyles(Grid)>
      </Block>
      <Block
        title="Current Monthly Rate"
      >
        <FancyDollarAmount>
          30
        </FancyDollarAmount>
        <Box
          my={25}
        >
          $5.00
          /person per month
        </Box>
        <styled.div>
          Total monthly rate does not include people covered by free trial conditions.
        </styled.div>
      </Block>
      <Block
        title="Current Billing Period"
      >
        <WithStyles(Grid)
          container={true}
          justify="space-between"
          spacing={16}
        >
          <WithStyles(Grid)
            item={true}
          >
            <styled.span>
              Start date:
            </styled.span>
          </WithStyles(Grid)>
          <WithStyles(Grid)
            item={true}
          >
            01/01/2015
          </WithStyles(Grid)>
        </WithStyles(Grid)>
        <WithStyles(Grid)
          container={true}
          justify="space-between"
          spacing={16}
        >
          <WithStyles(Grid)
            item={true}
          >
            <styled.span>
              End date:
            </styled.span>
          </WithStyles(Grid)>
          <WithStyles(Grid)
            item={true}
          >
            01/31/2015
          </WithStyles(Grid)>
        </WithStyles(Grid)>
      </Block>
    </WithStyles(Grid)>
  </Section>
</styled.div>
`;

exports[`BillingInformation in app billing enabled with freemium users not over the limit renders the trial information 1`] = `
<styled.div>
  <Section
    title="Billing Information"
  >
    <WithStyles(Grid)
      container={true}
      justify="space-between"
    >
      <Block
        title="Current Active People"
      >
        <WithStyles(Grid)
          alignItems="flex-end"
          container={true}
          justify="space-between"
        >
          <WithStyles(Grid)
            item={true}
          >
            <Typography
              variant="emphasis"
            >
              3
            </Typography>
          </WithStyles(Grid)>
          <WithStyles(Grid)
            item={true}
          />
        </WithStyles(Grid)>
      </Block>
      <Block
        title="Current Monthly Rate"
      >
        <FancyDollarAmount>
          0
        </FancyDollarAmount>
        <Box
          my={25}
        >
          $5.00
          /person per month
        </Box>
        <styled.div>
          Total monthly rate does not include people covered by free trial conditions.
        </styled.div>
      </Block>
      <Block
        title="Current Billing Period"
      >
        <WithStyles(Grid)
          container={true}
          justify="space-between"
          spacing={16}
        >
          <WithStyles(Grid)
            item={true}
          >
            <styled.span>
              Start date:
            </styled.span>
          </WithStyles(Grid)>
          <WithStyles(Grid)
            item={true}
          >
            01/01/2015
          </WithStyles(Grid)>
        </WithStyles(Grid)>
        <WithStyles(Grid)
          container={true}
          justify="space-between"
          spacing={16}
        >
          <WithStyles(Grid)
            item={true}
          >
            <styled.span>
              End date:
            </styled.span>
          </WithStyles(Grid)>
          <WithStyles(Grid)
            item={true}
          >
            01/31/2015
          </WithStyles(Grid)>
        </WithStyles(Grid)>
        <FreeTrial
          trialEndsAt={null}
          trialLabel="Free users"
          trialTitle="FREE"
          trialUsersCount={5}
          trialsUsedCount={3}
        />
      </Block>
    </WithStyles(Grid)>
  </Section>
</styled.div>
`;

exports[`BillingInformation in app billing enabled within trial, exceeded trial users renders with trial and billing information 1`] = `
<styled.div>
  <Section
    title="Billing Information"
  >
    <WithStyles(Grid)
      container={true}
      justify="space-between"
    >
      <Block
        title="Current Active People"
      >
        <WithStyles(Grid)
          alignItems="flex-end"
          container={true}
          justify="space-between"
        >
          <WithStyles(Grid)
            item={true}
          >
            <Typography
              variant="emphasis"
            >
              100
            </Typography>
          </WithStyles(Grid)>
          <WithStyles(Grid)
            item={true}
          >
            Total People
          </WithStyles(Grid)>
        </WithStyles(Grid)>
        <div>
          <styled.div>
            <Box
              my={20}
              p={10}
            >
              <WithStyles(Grid)
                container={true}
                justify="space-between"
              >
                <WithStyles(Grid)
                  item={true}
                >
                  5
                </WithStyles(Grid)>
                <WithStyles(Grid)
                  item={true}
                >
                  People on free trial
                </WithStyles(Grid)>
              </WithStyles(Grid)>
            </Box>
          </styled.div>
          <HorizontalDivider />
          <Box
            p={10}
          >
            <WithStyles(Grid)
              container={true}
              justify="space-between"
            >
              <WithStyles(Grid)
                data-jest="billableUserCount"
                item={true}
              >
                95
              </WithStyles(Grid)>
              <WithStyles(Grid)
                item={true}
              >
                Billable people
              </WithStyles(Grid)>
            </WithStyles(Grid)>
          </Box>
        </div>
      </Block>
      <Block
        title="Current Monthly Rate"
      >
        <FancyDollarAmount>
          475
        </FancyDollarAmount>
        <Box
          my={25}
        >
          $5.00
          /person per month
        </Box>
        <styled.div>
          Total monthly rate does not include people covered by free trial conditions.
        </styled.div>
      </Block>
      <Block
        title="Current Billing Period"
      >
        <WithStyles(Grid)
          container={true}
          justify="space-between"
          spacing={16}
        >
          <WithStyles(Grid)
            item={true}
          >
            <styled.span>
              Start date:
            </styled.span>
          </WithStyles(Grid)>
          <WithStyles(Grid)
            item={true}
          >
            01/01/2015
          </WithStyles(Grid)>
        </WithStyles(Grid)>
        <WithStyles(Grid)
          container={true}
          justify="space-between"
          spacing={16}
        >
          <WithStyles(Grid)
            item={true}
          >
            <styled.span>
              End date:
            </styled.span>
          </WithStyles(Grid)>
          <WithStyles(Grid)
            item={true}
          >
            01/31/2015
          </WithStyles(Grid)>
        </WithStyles(Grid)>
        <FreeTrial
          trialEndsAt="01/01/2022"
          trialLabel="Free trials used"
          trialTitle="FREE TRIAL"
          trialUsersCount={5}
          trialsUsedCount={5}
        />
      </Block>
    </WithStyles(Grid)>
  </Section>
</styled.div>
`;

exports[`BillingInformation organization is deactivated renders nothing 1`] = `""`;
